os/binfmt/binfmt_loadbinary.c

```diff
@@ -111,6 +123,15 @@ int load_binary(FAR const char *filename, size_t binsize, size_t offset)
 
 	sched_lock();
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	/* The first 4 bytes of the text section of the application must contain a
+	pointer to the application's mm_heap object. Here we will store the mm_heap
+	pointer to the start of the text section */
+	*(uint32_t *)(bin->alloc[0]) = (uint32_t)start_addr;
+	tcb = (struct tcb_s *)sched_self();
+	tcb->ram_start = (uint32_t)start_addr;
+#endif
+
 	/* Then start the module */
 
```

Comment:
Could you also add fault manager related code from binformat_exec.c here?
Please refer https://github.com/Samsung/TizenRT/pull/2836

Reply:
I think its not needed to merge the fault manager code, because it would be removed by the binary manager implementation.